ocfweb/docs/views/officers.py

```diff
@@ -15,39 +16,63 @@
 
 from ocfweb import caching
 
-_Term = namedtuple('_Term', ['name', 'gms', 'sms', 'dgms', 'dsms', 'heads'])
-Committee = namedtuple('Committee', ['name', 'heads'])
+OfficerUidOrInfo = Union[str, Tuple[str, date, date], Tuple[str, date, date, bool]]
 
 
-def Term(
-    name: str,
-    gms: List[Any],
-    sms: List[Any],
-    dgms: Optional[List[Any]] = None,
-    dsms: Optional[List[Any]] = None,
-    heads: Optional[List[Tuple[str, List[Any]]]] = None,
-) -> _Term:
-    gms = list(map(Officer.from_uid_or_info, gms))
-    sms = list(map(Officer.from_uid_or_info, sms))
-    dgms = list(map(Officer.from_uid_or_info, dgms or []))
-    dsms = list(map(Officer.from_uid_or_info, dsms or []))
-    heads = [
-        Committee(committee[0], list(map(Officer.from_uid_or_info, committee[1])))
-        for committee in heads or []
-    ]
-    return _Term(name, gms, sms, dgms, dsms, heads)
+@dataclass
+class Committee:
+    name: str
+    heads: 'List[Officer]'
+
+
+@dataclass
+class Term:
+    name: str
+    gms: 'List[Officer]'
+    sms: 'List[Officer]'
+    dgms: 'List[Officer]'
+    dsms: 'List[Officer]'
+    heads: List[Committee]
 
+    def __init__(
+        self,
+        name: str,
+        gms: List[OfficerUidOrInfo],
+        sms: List[OfficerUidOrInfo],
+        dgms: Optional[List[OfficerUidOrInfo]] = None,
+        dsms: Optional[List[OfficerUidOrInfo]] = None,
+        heads: Optional[List[Tuple[str, List[OfficerUidOrInfo]]]] = None,
+    ):
+        self.name = name
+        self.gms = list(map(Officer.from_uid_or_info, gms))
+        self.sms = list(map(Officer.from_uid_or_info, sms))
+        self.dgms = list(map(Officer.from_uid_or_info, dgms or []))
+        self.dsms = list(map(Officer.from_uid_or_info, dsms or []))
+        self.heads = [
+            Committee(committee[0], list(map(Officer.from_uid_or_info, committee[1])))
+            for committee in heads or []
+        ]
 
-class Officer(namedtuple('Officer', ['uid', 'name', 'start', 'end', 'acting'])):
+
+@dataclass
+class Officer:
+    uid: str
+    name: str
+    start: Optional[date]
+    end: Optional[date]
+    acting: bool
 
     @classmethod
-    def from_uid_or_info(cls: Callable[..., Any], uid_or_info: Union[Tuple[Any, ...], str]) -> Any:
+    def from_uid_or_info(cls: 'Callable[..., Officer]', uid_or_info: OfficerUidOrInfo) -> 'Officer':
+        start: Optional[date]
+        end: Optional[date]
+
         if isinstance(uid_or_info, tuple):
             if len(uid_or_info) == 3:
-                uid, start, end = uid_or_info
+                uid, start, end = cast(Tuple[str, date, date], uid_or_info)
```

Comment:
Perhaps this could be made a `typing.NamedTuple`?

Reply:
It seems that if I were to do that, I would have to add more boilerplate to the actual list of officers, right? I'm not sure I would like to do so at this time...

Reply:
Well, you could do something like `OfficerInfo(*uid_or_tupl)`, but thinking about it more, its probably not worth it.